sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1458,17 +1458,18 @@ static void _handle_rtr_timeout(_nib_dr_entry_t *router)
 {
     if ((router->next_hop != NULL) && (router->next_hop->mode & _DRL)) {
         _nib_offl_entry_t *route = NULL;
-        unsigned iface = _nib_onl_get_if(router->next_hop);
-        ipv6_addr_t addr = router->next_hop->ipv6;
+        _nib_onl_entry_t *next_hop = router->next_hop;
 
         _nib_drl_remove(router);
-        /* also remove all routes to that router */
+        /** The Router Lifetime applies only to
+            the router's usefulness as a default router; it
+            does not apply to information contained in other
+            message fields or options. Options that need time
+            limits for their information include their own
+            lifetime fields.
+            @see [RFC4861]https://datatracker.ietf.org/doc/html/rfc4861#section-4.2 */
         while ((route = _nib_offl_iter(route))) {
-            if ((route->next_hop != NULL) &&
-                (_nib_onl_get_if(route->next_hop) == iface) &&
-                (ipv6_addr_equal(&route->next_hop->ipv6, &addr))) {
-                route->mode = _EMPTY;
-                route->next_hop->mode &= ~_DST;
+            if (route->next_hop == next_hop) {
                 _nib_offl_clear(route);
```

Comment:
I guess this introduces some wrong usage of `_nib_offl_clear()` ;-) (see comment above)

Reply:
If `route->next_hop == router->next_hop` then 
`(_nib_onl_get_if(route->next_hop) == _nib_onl_get_if(router->next_hop))` and
`ipv6_addr_equal(&route->next_hop->ipv6, &router->next_hop->ipv6)`

Reply:
I understand the if, thank you. I was commenting on the removal of 

```
                route->mode = _EMPTY;
                route->next_hop->mode &= ~_DST;
```

Reply:
Ah ok.
> route->next_hop->mode &= ~_DST;

That's done in `_nib_drl_remove`.

>  route->mode = _EMPTY;

That enforces what `_nib_offl_clear()` did expect, but what I think is a bit too aggressive and causes trouble when an entry is not properly deleted.

> I'd argue: then don't call this function, but ok, let's go this route. But no guarantee if this introduces new bugs...

Maybe it is called when it should not be called. To find that is hard. If I change when it is called I think I would have to make more changes, which increases the probability for a new bug.
The `if (dst->mode == _EMPTY) {` prevents that in `tests/net/gnrc_ipv6_nib` a prefix list entry is not memsetted.
In `test_handle_router_timeout()`. It is not really an unexpected condition that the `if` is skipped. It is more a measure of safety, so an `assert()`would be misplaced but do you want me to add a little `DEBUG()`?